sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Triggers.scala

```diff
@@ -27,3 +32,34 @@ import org.apache.spark.sql.streaming.Trigger
 @Experimental
 @Evolving
 case object OneTimeTrigger extends Trigger
+
+/**
+ * A [[Trigger]] that runs a query periodically based on the processing time. If `interval` is 0,
+ * the query will run as fast as possible.
+ */
+@Evolving
+private[sql] case class ProcessingTimeTrigger(intervalMs: Long) extends Trigger {
+  require(intervalMs >= 0, "the interval of trigger should not be negative")
+}
+
+private[sql] object ProcessingTimeTrigger {
```

Comment:
Later we could make this non-private, but it seems like the intent is to expose through `Trigger.ProcessingTime` despite the fact that `OneTimeTrigger` is not private. Looks OK to me.

Reply:
Just a flyby comment (apologies if i am missing any prior context or conversation). I do not think it make sense to have both `OnTimeTrigger/ProcessingTimeTrigger` and `Trigger.OneTime/ProcessingTime` be public. The reason I had deprecated `ProcessingTime` constructor is so that we can reduce the number of public classes and move towards having everything through the `Trigger` class using`Trigger.ProcessingTime`. If we are indeed doing that, then we should have `Trigger.ProcessingTime` as the only public method and make `ProcessingTimeTrigger` private/hidden.

Reply:
Yeah it fits on my understanding (assumption, actually) of deprecation - let end users only know 
 about `Trigger.xxx` and use these methods instead of instantiating Trigger implementation by theirselves.